### ISSUE TITLE
Custom elements not loading using cellular network

### ISSUE THREAD

**Issue by andcarbo** [home-assistant/iOS#2977]
**iOS device model, version and app version**
Model Name: iPhone 14 Pro
Software Version: iOS 18.0 public beta (22A5350a) (also happening on iPhone 15 Pro w/iOS 17.6.1 (21G93))
App version: 2024.8.1 (2024.791)

**Home Assistant Core Version**
2024.9.0

**Describe the bug**
Using the companion app via cellular network results in very slow loading and a dashboard full of "custom element doesn't exist" cards (see screenshots). This happens with mushroom cards but also with any other custom element installed via HACS (layout-card, card-mod...). Also the entities state translation is not applied.
Happens since HA Core 2024.8.x (also updated HACS to v2.0 the same day).

**To Reproduce**
Load iOS companion app using cellular network

**Expected behavior**
Load cards as seen on second screenshot

**Screenshots**
Sample page loaded using cellular:
![IMG_747172226237](https://github.com/user-attachments/assets/aec651cf-9389-4cba-8efc-f81af6ce1ca8)

Same page loaded using local wi-fi:
![IMG_747172747170](https://github.com/user-attachments/assets/3b90ffb1-5465-427f-b33f-99876a8f1e21)

**Additional context**
Have a subscription, so when I'm not home I'm connecting via Nabu Casa remote control.
Everything works as expected when using local Wi-Fi connection.
Cards load correctly using Safari, either via wi-fi or cellular.
No problem loading cards on PC when not at home (using Chrome).
Using cellular, if I reload the resources manually (Settings > Dashboards > 3-dots) cards appear until I swipe down to reload the page (even if entities state translation is still not applied).

What I've tried:
- clearing the cache;
- resetting companion app (even though resetting via cellular doesn't work at all: loading about ten seconds then the app remains logged as if nothing happened);
- uninstalling and reinstalling multiple times;
- installing the latest beta from Testflight;
- removing resources and files from HACS relating the custom cards and reinstalling;
- deleting the session token;
- removing the device from the Mobile App integration;

**Comment by bgoncal** [comment]
How does it behave in iOS Safari using your Home assistant cloud URL? 

You said you cleared cache already, you mean via companion app settings >> debugging >> clear frontend cache?

**Comment by andcarbo** [comment]
Yes, cleared cache via companion app settings -> debug -> clear frontend cache.
Here is how it looks using safari on cellular (only slightly different because of the zoom level):
![IMG_747175224287](https://github.com/user-attachments/assets/94ef92d5-8265-4b3d-b04e-fbca14028bcb)

**Comment by bgoncal** [comment]
Weird, it does still look some sort of frontend cache to me, do you have another iOS device to test to see if it is just on your iPhone that the issue happens?

**Comment by andcarbo** [comment]
I just tried clearing cache on the other iPhone (15 Pro with iOS non-beta 17.6.1), but nothing has changed

**Comment by bgoncal** [comment]
Please export your logs (Companion app settings >> debugging >> export logs) and submit here: https://forms.gle/Wbsmbdn4CHfCDd7e9

**Comment by andcarbo** [comment]
Ok, done. Thanks for your support!
